### PR TITLE
Add first draft of documentation

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,26 +2,89 @@
 
 [![Version](https://img.shields.io/maven-central/v/io.quarkiverse.argocd/quarkus-argocd?logo=apache-maven&style=flat-square)](https://central.sonatype.com/artifact/io.quarkiverse.argocd/quarkus-argocd-parent)
 
-## Welcome to Quarkiverse!
+## Features
 
-Congratulations and thank you for creating a new Quarkus extension project in Quarkiverse!
+- Generate the ArgoCD Application Kubernetes CR for the Quarkus application
+- Command Line interface to install / unistall and list Application(s)
```
Comment (cmoulliard) [inline README.md:8]: `unistall` --> `uninstall`

File: README.md [hunk 2]
```diff
@@ -2,26 +2,89 @@
...
+- Integration with Quarkus Helm
 
-Feel free to replace this content with the proper description of your new project and necessary instructions how to use and contribute to it.
 
-You can find the basic info, Quarkiverse policies and conventions in [the Quarkiverse wiki](https://github.com/quarkiverse/quarkiverse/wiki).
+## Building
 
-In case you are creating a Quarkus extension project for the first time, please follow [Building My First Extension](https://quarkus.io/guides/building-my-first-extension) guide.
+To build the extension use the following command:
```
Comment (cmoulliard) [inline README.md:14]: Maybe use `To build this extension which is also a plugin, use ....`

File: README.md [hunk 3]
```diff
@@ -2,26 +2,89 @@
...
+## Usage
 
-## Documentation
+To get the the argocd custom resources generated one needs to add the `quarkus-argocd` extension to the project.
 
-The documentation for this extension should be maintained as part of this repository and it is stored in the `docs/` directory.
+### Add extension to your project 
 
-The layout should follow the [Antora's Standard File and Directory Set](https://docs.antora.org/antora/2.3/standard-directories/).
+To add the extension to the project you to manually edit the `pom.xml` / `build.gradle` file.
```
Comment (cmoulliard) [inline README.md:26]: `To use the extension to a Quarkus project, manually edit the ....`

File: README.md [hunk 4]
```diff
@@ -2,26 +2,89 @@
...
+
+#### Manually editing the `build.gradle` file
+
+```groovy
+dependencies {
+    implementation 'io.quarkiverse.argocd:quarkus-argocd:999-SNAPSHOT'
+}
+```
+
+After this step the ArgoCD CR will be generated under `.argocd` directory in the project root.
```
Comment (cmoulliard) [inline README.md:46]: `After this step` assumes that you build locally your code ;-)

File: README.md [hunk 5]
```diff
@@ -2,26 +2,89 @@
...
+
+
+### Using the CLI
+
+The project provides a companion CLI that can be used to install / uninstall and list the applications.
```
Comment (cmoulliard) [inline README.md:51]:
```
....list the Argocd Applications.
```

File: README.md [hunk 6]
```diff
@@ -2,26 +2,89 @@
...
+The CLI can be added with thw following command:
```
Comment (cmoulliard) [inline README.md:52]: `thw` -> `the`

File: README.md [hunk 7]
```diff
@@ -2,26 +2,89 @@
...
+
+To re-triggger the file genration:
+
+```shell
+quarkus argocd application generate
+```
+
+#### Installing the application
+
+To install generated `Application` CR to the currently connected Kubernetes cluster:
```
Comment (cmoulliard) [inline README.md:68]: Prior to use the one of the Argocd commands, you should mention in a pre-req section that an active connection to a Kubernetes cluster where Argocd has been installed are needed and that the user knows which version of argocd is supported

File: README.md (hunk 2 above)
Comment (iocanel) [reply README.md:14]: We've already mentioned that the project provides a plugin, above

**Pull request merged**